tic_tac_toe.rb

```diff
@@ -4,78 +4,46 @@ def initialize(board)
   end
 
   def winner
-    row1 = @board[0]
-    row2 = @board[1]
-    row3 = @board[2]
-
-    # row checks
-
-    if row1[0] == "o" && row1[1] == "o" && row1[2] == "o"
-      return "o"
-    end
-
-    if row2[0] == "o" && row2[1] == "o" && row2[2] == "o"
-      return "o"
-    end
-
-    if row3[0] == "o" && row3[1] == "o" && row3[2] == "o"
-      return "o"
-    end
-
-    if row1[0] == "x" && row1[1] == "x" && row1[2] == "x"
-      return "x"
-    end
-
-    if row2[0] == "x" && row2[1] == "x" && row2[2] == "x"
-      return "x"
-    end
-
-    if row3[0] == "x" && row3[1] == "x" && row3[2] == "x"
-      return "x"
-    end
-
-    # column checks
-
-    if row1[0] == "o" && row2[0] == "o" && row3[0] == "o"
-      return "o"
-    end
-
-    if row1[1] == "o" && row2[1] == "o" && row3[1] == "o"
-      return "o"
+    # rows
+    @board.each do |row|
+      return check_row(row) if check_row(row)
     end
```

Comment:
Hi! This is Daniel Ramirez from ApproveShield :)

After reading your solution I found this pattern repeated a couple times, which results in the row being
checked for a winner twice. Do you think there is a better approach that might avoid this issue?

Reply:
Hey Daniel, 

I noticed this clunky bit of code, but I couldn't quite figure out a 'good' way to solve it. 

As I understand it, an explicit return is necessary to end the program and prevent it from checking for more winners after one has already been found.  The if statement is also necessary to prevent a return if there's no winner.

It's possible to remove `return check_row(row) if check_row(row)` entirely and replace it with the code of `check_row`: (similar story with `check_diags`)
```
winning_char = row.uniq
return winning_char.first if winning_char.length == 1 && winning_char.first != " "
```
Which is more efficient, but I'd also be repeating code several times, and I wanted to keep it dry.

I'm sure there is a better way, but maybe that's it; I'm definitely open to other possibilities.